Mathlib/CategoryTheory/Preadditive/Injective/Preserves.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2025 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import Mathlib.CategoryTheory.Preadditive.Injective.Basic
+
+/-!
+# Preservation of injective objects
+
+We define a typeclass `Functor.PreservesInjectiveObjects`.
+
+We restate the existing result that if `F ⊣ G` is an adjunction and `F` preserves monomorphisms,
+then `G` preserves injective objects. We show that the converse is true if the codomain of `F` has
+enough injectives.
+-/
+
+universe v₁ v₂ v₃ u₁ u₂ u₃
+
+namespace CategoryTheory
+
+variable {C : Type u₁} [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D]
+  {E : Type u₃} [Category.{v₃} E]
+
+/-- A functor preserves injective objects if it maps injective objects to injective objects. -/
+class Functor.PreservesInjectiveObjects (F : C ⥤ D) : Prop where
+  injective_obj {X : C} : Injective X → Injective (F.obj X)
+
+/-- See `Functor.injective_obj_of_injective` for a variant taking `Injective X` as an explicit
+argument. -/
+instance Functor.injective_obj (F : C ⥤ D) [F.PreservesInjectiveObjects] (X : C) [Injective X] :
+    Injective (F.obj X) :=
+  Functor.PreservesInjectiveObjects.injective_obj inferInstance
+
+/-- See `Functor.injective_obj` for a variant taking `Injective X` as a typeclass argument. -/
+theorem Functor.injective_obj_of_injective (F : C ⥤ D) [F.PreservesInjectiveObjects] {X : C}
+    (h : Injective X) : Injective (F.obj X) :=
+  Functor.PreservesInjectiveObjects.injective_obj h
+
+theorem Functor.preservesInjectiveObjects_comp (F : C ⥤ D) (G : D ⥤ E)
```

Comment:
I think this one should also be an instance, as well as `Functor.preservesInjectiveObjects_of_isEquivalence`.

Reply:
Done.